pkg/pb/metainfo.proto

```diff
@@ -328,4 +340,139 @@ message SatStreamID {
+
+    bytes encrypted_inline_data = 6;
+    repeated Piece pieces = 7;
+}
+
+message Piece {
+    int32 piece_num = 1;
+    bytes node = 2[(gogoproto.customtype) = "NodeID", (gogoproto.nullable) = false];
+}
+
+message SegmentPosition {
+    int32 part_number = 1;
+    int32 index = 2;
+}
+
+message SegmentBeginRequest {
+    bytes           stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+    SegmentPosition position = 2;
+
+    int64 max_order_limit = 3;
+}
+
+message SegmentBeginResponse {
+    bytes    segment_id = 1 [(gogoproto.customtype) = "SegmentID", (gogoproto.nullable) = false];
+    repeated AddressedOrderLimit addressed_limits = 2;
+    bytes private_key = 3 [(gogoproto.customtype) = "PiecePrivateKey", (gogoproto.nullable) = false];
+}
+
+message SegmentCommitRequest {
+    bytes segment_id = 1 [(gogoproto.customtype) = "SegmentID", (gogoproto.nullable) = false];
+
+    bytes encrypted_key_nonce = 2 [(gogoproto.customtype) = "Nonce", (gogoproto.nullable) = false];
+    bytes encrypted_key = 3;
+
+    int64 size_encrypted_data = 4; // refers to segment size not piece size
+
+    repeated SegmentPieceUploadResult upload_result = 5;
+}
+
+message SegmentPieceUploadResult {
+    int32               piece_num = 1;
+    bytes               node_id = 2 [(gogoproto.customtype) = "NodeID", (gogoproto.nullable) = false];
+    orders.PieceHash    hash = 3;
+}
+
+// only for satellite use
+message SatSegmentID {
+    SatStreamID stream_id = 1;
+    int32    part_number = 2;
+    int32    index = 3;
+
+    // TODO we have redundancy in SatStreamID, do we need it here?
+    // pointerdb.RedundancyScheme redundancy = 4;
+    bytes root_piece_id = 5 [(gogoproto.customtype) = "PieceID", (gogoproto.nullable) = false];
+    repeated AddressedOrderLimit original_order_limits = 6;
+    google.protobuf.Timestamp creation_date = 7  [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
+
+    bytes satellite_signature = 8;
+}
+
+message SegmentCommitResponse {}
+
+message SegmentMakeInlineRequest {
+    bytes stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+    SegmentPosition position = 2;
+
+    bytes encrypted_key_nonce = 3 [(gogoproto.customtype) = "Nonce", (gogoproto.nullable) = false];
+    bytes encrypted_key = 4;
+
+    bytes encrypted_inline_data = 5;
+}
+
+message SegmentMakeInlineResponse {}
+
+message SegmentBeginDeleteRequest {
+    bytes stream_id = 1;
+    SegmentPosition position = 2;
+}
+
+message SegmentBeginDeleteResponse {
+    bytes segment_id = 1 [(gogoproto.customtype) = "SegmentID", (gogoproto.nullable) = false];
+    repeated AddressedOrderLimit addressed_limits = 2;
+}
+
+message SegmentFinishDeleteRequest {
+    bytes segment_id = 1 [(gogoproto.customtype) = "SegmentID", (gogoproto.nullable) = false];
+    repeated SegmentPieceDeleteResult results = 2;
+}
+
+message SegmentPieceDeleteResult {
+    int32               piece_num = 1;
```

Comment:
Is this message sent by satellite to uplink,? what causes this message to be generated? how is this message handled? is this a request or response message?

Reply:
its part of `SegmentFinishDeleteRequest`, it will be sent from uplink to satellite after deleting pieces from storage nodes